web-app/test/unit/resources/elements/panels/main-panel-spec.js

```diff
@@ -114,34 +114,80 @@ describe('MainPanel', () => {
 
     describe('setScalingFactor', () => {
 
+        const dataUri = 'data:image/jpeg;base64,R0lGODlhkAGQAfcAAAQCBJKDLJSSlExGHNS/NNTOtFRSVCckFGxjJPjgXMzKzHyCfKuqpDMxNPzu0fXaJGxqZLikLEdCPBQSDDQyFOvq7H90LL6PlFxOw==';
+
```

Comment:
STEP1: This is all a little complicated but I will try and explain.  We create a mock object with a src property and onload function.  We save OFF the "Real" image object/prototype for later.  Then we replace the global (aka Window - but jest handles window a little differently) image with our fake image (and make sure we restore it after all these tests - otherwise other test classes/suites might use this mock image!)